pkg/scheduler/internal/cache/cache.go

```diff
@@ -29,6 +29,7 @@ import (
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	"k8s.io/kubernetes/pkg/features"
 	"k8s.io/kubernetes/pkg/scheduler/algorithm"
+	"k8s.io/kubernetes/pkg/scheduler/metrics"
```

Comment:
It looks like not all cache size change events are captured, e.g,, Line #400 in original file where `delete(cache.assumedPods, key)`. 

I think the best way to capture everything is to go through all references of the 3 maps `assumedPods`, `podStates` and `nodes`, and `Inc/Dec` the metric when `add/delete` happens. There might be an opportunity to refactor the code as well, e.g., if we make all `add/delete` operation go through a single set of helper functions, then we only need to `Inc/Dec` in those helper functions.

Reply:
That's a good point, I also think there is a chance to refactor some of the cache code here. Will work more on it

Reply:
it seems error prone to Inc/Dec because I am pretty sure we will forget a place and the counter will diverge from the actual cache size and can never recover, how about we have a function that ```Set``` the value of the metrics to ```len(map)``` of each map, instead of doing inc/dec